common/routes/index.jsx

```diff
@@ -1,25 +1,35 @@
 import React from 'react'
 import {Route, IndexRoute} from 'react-router'
-import {routerActions} from 'react-router-redux'
 import {UserAuthWrapper as userAuthWrapper} from 'redux-auth-wrapper'
+import {push} from 'react-router-redux'
 
 // TODO: use webpack code-splitting and System.import to reduce initial bundle size
+import {userCan} from 'src/common/util'
+import {authorizationError} from 'src/common/actions/errors'
 import App from 'src/common/containers/App'
-
 import BlankLayout from 'src/common/containers/BlankLayout'
 import SignUp from 'src/common/containers/SignUp'
 import SignIn from 'src/common/containers/SignIn'
-
 import Home from 'src/common/containers/Home'
 import Profile from 'src/common/components/Profile'
+import UsersContainer from 'src/common/containers/Users'
 import {buildURL} from 'src/common/util'
 
-const userIsAuthenticated = userAuthWrapper({
-  failureRedirectPath: '/sign-in',
-  authSelector: state => state.auth.currentUser,
-  redirectAction: routerActions.replace,
-  wrapperDisplayName: 'userIsAuthenticated',
-})
+const userCanVisit = (capability, store) => {
+  return userAuthWrapper({
+    authSelector: state => state.auth.currentUser,
+    predicate: currentUser => userCan(currentUser, capability),
+    failureRedirectPath: '/sign-in',
```

Comment:
Like echo, should redirect to `/not-found` instead of to `/sign-in`. See comments above re: authentication.